homeassistant/components/switch/freebox.py

```diff
@@ -0,0 +1,103 @@
+https://home-assistant.io/components/switch.freebox/
+"""
+import logging
+
+from homeassistant.components.freebox import DATA_FREEBOX
+from homeassistant.const import (STATE_OFF, STATE_ON)
+from homeassistant.helpers.entity import ToggleEntity
+
+DEPENDENCIES = ['freebox']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the sensors."""
+    fbx = hass.data[DATA_FREEBOX]
+    perms_settings = discovery_info.get('perms_settings')
+    async_add_entities([
+        FbxWifiSwitch(fbx, perms_settings)
+    ], True)
+
+
+class FbxWifiSwitch(ToggleEntity):
+    """Representation of a freebox wifi switch."""
+
+    def __init__(self, fbx, perms_settings):
+        """Initilize the Wifi switch."""
+        self._name = 'Freebox WiFi'
+        self._state = None
+        self.perms_settings = perms_settings
+        self.fbx = fbx
+
+    @property
+    def available(self):
+        """If permission is not true the switch is not available."""
+        if not self.perms_settings:
+            return False
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the switch."""
+        return self._state
+
+    @property
+    def should_poll(self):
+        """Poll for status."""
+        return True
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state == STATE_ON
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the switch on."""
+        from aiofreepybox.exceptions import InsufficientPermissionsError
+
+        wifi_config = {"enabled": True}
+        try:
+            await self.fbx.wifi.set_global_config(wifi_config)
+        except InsufficientPermissionsError:
+            _LOGGER.warning('Home Assistant does not have permissions to'
+                            ' modify the settings. Please refer to'
+                            ' documentation. https://home-assistant.io/'
+                            'components/switch.freebox/')
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the switch off."""
+        from aiofreepybox.exceptions import InsufficientPermissionsError
+
+        wifi_config = {"enabled": False}
+        try:
+            await self.fbx.wifi.set_global_config(wifi_config)
+        except InsufficientPermissionsError:
+            _LOGGER.warning('Home Assistant does not have permissions to'
+                            ' modify the settings. Please refer to'
+                            ' documentation. https://home-assistant.io/'
+                            'components/switch.freebox/')
+
+    async def async_update(self):
+        """Get the state and update it."""
+        from aiofreepybox.constants import PERMISSION_SETTINGS
+
+        permissions = await self.fbx.get_permissions()
```

Comment:
It seems inefficient if each switch will query the permissions on each update. That needs to be shared.

Reply:
I understand the comment (i think) but i don't know how to change this. I need to check the permission status to make it available or not without HA restart and the 1st permission check in ../freebox.py component.
If I understand correct your comment you ask to get permission globally for all switches and then check if the required permission is OK in each switch update (even there's only 1 switch at the moment) ?

Reply:
I think i got it...